tests/lib/ringbuffer/src/main.c

```diff
@@ -227,6 +230,167 @@ void test_ringbuffer_size_put_get_thread_isr(void)
+{
+	int i;
+	u8_t inbuf[RINGBUFFER_SIZE];
+	u8_t outbuf[RINGBUFFER_SIZE];
+	size_t in_size;
+	size_t out_size;
+
+	/* Initialize test buffer. */
+	for (i = 0; i < RINGBUFFER_SIZE; i++) {
+		inbuf[i] = i;
+	}
+
+	for (i = 0; i < 10; i++) {
+		memset(outbuf, 0, sizeof(outbuf));
+		in_size = ring_buf_put(&ringbuf_raw, inbuf,
+					       RINGBUFFER_SIZE - 2);
+		out_size = ring_buf_get(&ringbuf_raw, outbuf,
+						RINGBUFFER_SIZE - 2);
+
+		zassert_true(in_size == RINGBUFFER_SIZE - 2, NULL);
+		zassert_true(in_size == out_size, NULL);
+		zassert_true(memcmp(inbuf, outbuf, RINGBUFFER_SIZE - 2) == 0,
+			     NULL);
+	}
+
+	in_size = ring_buf_put(&ringbuf_raw, inbuf,
+				       RINGBUFFER_SIZE);
+	zassert_equal(in_size, RINGBUFFER_SIZE - 1, NULL);
+
+	in_size = ring_buf_put(&ringbuf_raw, inbuf,
+				       1);
+	zassert_equal(in_size, 0, NULL);
+
+	out_size = ring_buf_get(&ringbuf_raw, outbuf,
+					RINGBUFFER_SIZE);
+
+	zassert_true(out_size == RINGBUFFER_SIZE - 1, NULL);
+
+	out_size = ring_buf_get(&ringbuf_raw, outbuf,
+					RINGBUFFER_SIZE + 1);
+	zassert_true(out_size == 0, NULL);
+
+
+}
+
+void test_ringbuffer_alloc_put(void)
+{
+	u8_t outputbuf[RINGBUFFER_SIZE];
+	u8_t inputbuf[] = {1, 2, 3, 4};
+	u32_t read_size;
+	u32_t allocated;
+	u32_t sum_allocated;
+	u8_t *data;
+	int err;
+
+	ring_buf_init(&ringbuf_raw, RINGBUFFER_SIZE, ringbuf_raw.buf.buf8);
+
+	allocated = ring_buf_put_claim(&ringbuf_raw, &data, 1);
+	sum_allocated = allocated;
+	zassert_true(allocated == 1, NULL);
+
+
+	allocated = ring_buf_put_claim(&ringbuf_raw, &data,
+					   RINGBUFFER_SIZE - 1);
+	sum_allocated += allocated;
+	zassert_true(allocated == RINGBUFFER_SIZE - 2, NULL);
+
+	/* Putting too much returns error */
+	err = ring_buf_put_finish(&ringbuf_raw, RINGBUFFER_SIZE);
+	zassert_true(err != 0, NULL);
+
+	err = ring_buf_put_finish(&ringbuf_raw, 1);
+	zassert_true(err == 0, NULL);
+
+	err = ring_buf_put_finish(&ringbuf_raw, RINGBUFFER_SIZE - 2);
+	zassert_true(err == 0, NULL);
+
+	read_size = ring_buf_get(&ringbuf_raw, outputbuf,
+					     RINGBUFFER_SIZE - 1);
+	zassert_true(read_size == (RINGBUFFER_SIZE - 1), NULL);
+
+	for (int i = 0; i < 10; i++) {
+		allocated = ring_buf_put_claim(&ringbuf_raw, &data, 2);
+		if (allocated == 2) {
+			data[0] = inputbuf[0];
+			data[1] = inputbuf[1];
+		} else {
+			data[0] = inputbuf[0];
+			ring_buf_put_claim(&ringbuf_raw, &data, 1);
+			data[0] = inputbuf[1];
+		}
+
+		allocated = ring_buf_put_claim(&ringbuf_raw, &data, 2);
+		if (allocated == 2) {
+			data[0] = inputbuf[2];
+			data[1] = inputbuf[3];
+		} else {
+			data[0] = inputbuf[2];
+			ring_buf_put_claim(&ringbuf_raw, &data, 1);
+			data[0] = inputbuf[3];
+		}
+
+		err = ring_buf_put_finish(&ringbuf_raw, 4);
+		zassert_true(err == 0, NULL);
+
+		read_size = ring_buf_get(&ringbuf_raw,
+						     outputbuf, 4);
+		zassert_true(read_size == 4, NULL);
+
+		zassert_true(memcmp(outputbuf, inputbuf, 4) == 0, NULL);
+	}
+}
+
+void test_raw_put_free(void)
```

Comment:
shouldn't this be `test_byte_put_free` ?

Reply:
yep, renamed.